### DISCUSSION TITLE
Slideshow doesn't load new pictures

### DISCUSSION THREAD

**Discussion by DeadstrikeGhost** [photoprism/photoprism#2972] [Q&A]
When I start a slideshow in a folder, it starts repeating after all the pictures that are currently loaded on the page have been shown. Is there a setting or something that I need to adjust to enable the slideshow to load in new pictures?

**Comment by lastzero** [comment]
What version are you using? It should preload a lot of pictures, basically all for reasonably sized albums. Only if albums are extremely large, the additional pictures can't be loaded dynamically by the current viewer. The new viewer will support that.

**Reply by DeadstrikeGhost** [reply]
photoprism/photoprism       latest            e6f531bbf1b0   2 weeks ago     1.5GB

What is the definition of "a lot of pictures". we currently have 5890 pictures that we want the slideshow to show. 
When is the new viewer coming out?

**Reply by lastzero** [reply]
As soon as I have time for it (currently working on multi-user support) or we can afford to hire developers. That depends on how many people support our work, so it's not up to us.

**Reply by DeadstrikeGhost** [reply]
Is there a workaround that I can implement for now?

**Reply by lastzero** [reply]
You could change the source and preload more I guess. However, that may break the viewer or consume a lot of memory which is why there is a limit.

**Reply by lastzero** [reply]
https://github.com/photoprism/photoprism/blob/6d3d1d18881c892a091adbfe244c9bbae0708b1a/frontend/src/page/photos.vue#L131

https://github.com/photoprism/photoprism/blob/6d3d1d18881c892a091adbfe244c9bbae0708b1a/frontend/src/model/photo.js#L89

**Reply by lastzero** [reply]
The number of preloaded pictures has been increased. Can you please test this with the development preview we provide?
- https://github.com/photoprism/photoprism/issues/3009
- https://docs.photoprism.app/release-notes/#development-preview

**Comment by arendtio** [comment]
Today, I saw #3009 in the release notes and eagerly installed the update, because I have a very similar issue. Sadly simply increasing the batch size is barely noticeable. I understand that this is not an urgent issue and that multi-user support has priority. However, I would like to share an example and some numbers.

My example is very basic: My son wants to see the pictures of the zoo we visited last October.

When I open the search (no filters, 'newest first') the request to get another batch takes about 180ms via the local network. So I add the filters for 2022 and October. Now the time rises to about 220ms per batch. But because I want to see them in the right order, I have to switch to 'oldest first'. Now another batch takes about 3.300ms. Since the zoo visit was at the end of October, I have to scroll down and down again until all photos are loaded. I took about 2.300 photos that month, so just manually scrolling until all are loaded takes about 2-3 minutes of manual work ;-)

My conclusion:

- First of all, I wonder why 'oldest first' has such a huge impact (factor x10 sounds weird when the difference shouldn't be measurable).
- In addition, I wonder if it is the correct approach to always do a 'search' when loading more images. Wouldn't it be better to send the client an id list of the complete search result at the start (maybe page it with a size of >10k) and let it request details for a list of photo ids? That way the complexity of the search would not affect the lazy loading performance.
- Having a filter based on actual dates (from, to) might improve the situation too

This might sound overly critical, so don't get me wrong. I am super impressed by the performance PhotoPrism delivers in general. But to improve it even further I wanted to give a specific example and share some ideas on how the situation could be improved. Maybe some of these ideas might find their way to the new viewer.

**Reply by lastzero** [reply]
Thanks! You could filter results with "after" and "before", see docs.photoprism.app. What database are you using? How many pictures in total? 3300ms for 2300 results seems odd indeed. 

While we may add caching and other improvements (e.g. a new viewer) in the future, we unfortunately don't have the resources required at this time.

**Reply by arendtio** [reply]
The `before` and `after` filters help a lot. I didn't know that the filters have their own little language 🚀

There are about 111.000 search results overall and about 174.000 original files. All on a Raspberry Pi 4B with an external HDD. So not exactly a best-case scenario hardware-wise, but in general, the system feels very good :+1: 

Thanks for the help.
